### DISCUSSION TITLE
My --pc variable have not the same value as documentation theme tool

### DISCUSSION THREAD

**Discussion by Suniron** [saadeghi/daisyui#1673] [Q&A] [answered]
Hello,

To test, I set my colors in the theme tool and I havn't the same text color on primary button :(

The values for `--pc` variable is not the same between my app and the doc 😪, why ?

Thanks ! :)

**Comment by saadeghi** [comment]
What is the text color you set and what is the CSS variable you get?  

Can you share your `tailwind.config.js`?

**Reply by Suniron** [reply]
Sure,

This is the config of my app:

```
const baseConfig = require('@myapp/tailwind/tailwind.config')

/** @type {import('tailwindcss').Config} */
const config = {
  ...baseConfig,
  content: [
    './components/**/*.js',
    './components/**/*.ts',
    './components/**/*.vue',
    './layouts/**/*.vue',
    './pages/**/*.vue',
    './plugins/**/*.js',
    './plugins/**/*.ts',
    './nuxt.config.ts',
    './app.vue',
    './node_modules/@myapp/ui/components/**/*.vue',
    './../../packages/ui/components/**/*.vue',
  ],
}

module.exports = config
```

And this one is my shared config (which import `DaisyUI)`:

```
/** @type {import('tailwindcss').Config} */
const config = {
  daisyui: {
    themes: [
      {
        light: {
          'accent': '#EA580C',
          'base-100': '#F8F9F9',
          'error': '#DD0800',
          'info': '#DEF2FF',
          'neutral': '#D0D1D1',
          'primary': '#00B53B',
          'secondary': '#4CA591',
          'success': '#D4F9D8',
          'warning': '#FBBD23',
        },
      },
    ],
  },
  plugins: [require('daisyui')],
}

module.exports = config
```

**Comment by Suniron** [comment, marked as answer]
I just set the classes `btn btn-primary` on my button and I see the `--pc` value is `139.56 100% 7.098%;` (vs `140 100% 87%;` in the documentation with my theme)

**Reply by saadeghi** [reply]
Got it. It is happening because of the theme-generator trims extra decimal places when converting HEX to HSL. And that small decimal place for this special color `#00B53B` makes the difference if the content color should be light or dark.  
I will fix this on theme generator soon.  

In the meanwhile, set the color you want for the `primary-content` color (or [other optional colors](https://daisyui.com/docs/colors/)) like this:

```diff
  daisyui: {
    themes: [
      {
        light: {
          'accent': '#EA580C',
          'base-100': '#F8F9F9',
          'error': '#DD0800',
          'info': '#DEF2FF',
          'neutral': '#D0D1D1',
          'primary': '#00B53B',
+         'primary-content': '#FFF',
          'secondary': '#4CA591',
          'success': '#D4F9D8',
          'warning': '#FBBD23',
        },
      },
    ],
  },
```

**Reply by Suniron** [reply]
Oh GG I'll try this setting soon because the text color have not a really good contrast on a primary button with my config 😅.
